lib/httparrot/p_handler.ex

```diff
@@ -57,4 +58,70 @@ defmodule HTTParrot.PHandler do
   defp response(info, body) do
     info ++ body |> JSEX.encode!
   end
+
+  def post_multipart(req, _state) do
+    {:ok, parts, req} = handle_multipart(req)
+
+    filter = fn({type, _name, _body}, type_atom) -> type == type_atom end
+    reducer = fn({_type, name, body}, acc) -> acc ++ [{name, body}] end
+
+    # the other post handlers return a list with a single empty tuple if
+    # there's no data for forms, so let's match that behavior...
+    normalize = fn(parts) -> if parts == [], do: [{}], else: parts end
```

Comment:
Are you referring to post_multipart, or the filter,reducer, normalize anonymous fn's?